docs/modules/_ipfs_config_.md
[Fission SDK](../README.md) › ["ipfs/config"](_ipfs_config_.md)

# Module: "ipfs/config"

## Index

### Variables

* [DELEGATE_ADDR](_ipfs_config_.md#const-delegate_addr)
* [JS_IPFS](_ipfs_config_.md#const-js_ipfs)
* [PEER_WSS](_ipfs_config_.md#const-peer_wss)
* [SIGNALING_SERVER](_ipfs_config_.md#const-signaling_server)

### Functions

* [get](_ipfs_config_.md#const-get)
* [set](_ipfs_config_.md#const-set)

### Object literals

* [defaultOptions](_ipfs_config_.md#const-defaultoptions)

## Variables

### `Const` DELEGATE_ADDR

• **DELEGATE_ADDR**: *"/dns4/ipfs.runfission.com/tcp/443/https"* = "/dns4/ipfs.runfission.com/tcp/443/https"

*Defined in [src/ipfs/config.ts:15](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L15)*

___

### `Const` JS_IPFS

• **JS_IPFS**: *"https://cdnjs.cloudflare.com/ajax/libs/ipfs/0.50.1/index.min.js"* = "https://cdnjs.cloudflare.com/ajax/libs/ipfs/0.50.1/index.min.js"

*Defined in [src/ipfs/config.ts:12](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L12)*

___

### `Const` PEER_WSS

• **PEER_WSS**: *"/dns4/node.fission.systems/tcp/4003/wss/p2p/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw"* = "/dns4/node.fission.systems/tcp/4003/wss/p2p/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw"

*Defined in [src/ipfs/config.ts:13](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L13)*

___

### `Const` SIGNALING_SERVER

• **SIGNALING_SERVER**: *"/dns4/webrtc.runfission.com/tcp/443/wss/p2p-webrtc-star/"* = "/dns4/webrtc.runfission.com/tcp/443/wss/p2p-webrtc-star/"

*Defined in [src/ipfs/config.ts:14](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L14)*

## Functions

### `Const` get

▸ **get**(): *Promise‹IPFS›*

*Defined in [src/ipfs/config.ts:38](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L38)*

**Returns:** *Promise‹IPFS›*

___

### `Const` set

▸ **set**(`userIpfs`: unknown): *void*

*Defined in [src/ipfs/config.ts:34](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L34)*

**Parameters:**

Name | Type |
------ | ------ |
`userIpfs` | unknown |

**Returns:** *void*

## Object literals

### `Const` defaultOptions

### ▪ **defaultOptions**: *object*

*Defined in [src/ipfs/config.ts:21](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L21)*

▪ **config**: *object*

*Defined in [src/ipfs/config.ts:22](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L22)*

* **Bootstrap**: *string[]* = [ PEER_WSS ]

* **Addresses**: *object*

  * **Delegates**: *string[]* = [ DELEGATE_ADDR ]

  * **Swarm**: *string[]* = [ SIGNALING_SERVER ]

▪ **preload**: *object*

*Defined in [src/ipfs/config.ts:29](https://github.com/fission-suite/webnative/blob/693f51f/src/ipfs/config.ts#L29)*

* **enabled**: *boolean* = false


docs/modules/_keystore_config_.md
[Fission SDK](../README.md) › ["keystore/config"](_keystore_config_.md)

# Module: "keystore/config"

## Index

### Functions

* [clear](_keystore_config_.md#const-clear)
* [get](_keystore_config_.md#const-get)
* [set](_keystore_config_.md#const-set)

## Functions

### `Const` clear

▸ **clear**(): *Promise‹void›*

*Defined in [src/keystore/config.ts:11](https://github.com/fission-suite/webnative/blob/693f51f/src/keystore/config.ts#L11)*

**Returns:** *Promise‹void›*

___

### `Const` get

▸ **get**(): *Promise‹RSAKeyStore›*

*Defined in [src/keystore/config.ts:22](https://github.com/fission-suite/webnative/blob/693f51f/src/keystore/config.ts#L22)*

**Returns:** *Promise‹RSAKeyStore›*

___

### `Const` set

▸ **set**(`userKeystore`: RSAKeyStore): *Promise‹void›*

*Defined in [src/keystore/config.ts:18](https://github.com/fission-suite/webnative/blob/693f51f/src/keystore/config.ts#L18)*

**Parameters:**

Name | Type |
------ | ------ |
`userKeystore` | RSAKeyStore |

**Returns:** *Promise‹void›*


CHANGELOG.md
# Changelog

### v0.19.7

Bugfix: updates to files on public side were failing

### v0.19.6

Bugfix: changes to public tree were not being reflected in pretty tree

### v0.19.5

Don't error on failed pins

### v0.19.2

Do not recursively pin content

### v0.19.1

Permissions should be optional for `redirectToLobby` and `loadFileSystem` as well.

### v0.19

- Reliability & performance improvements
- Permissions are now optional



### v0.18.1

Added proofs to JWT for app routes (index, create & delete)

### v0.18.0

###### Breaking changes

- Renamed `publicise`/`publicize` to `publish`
- Renamed `prerequisites` to `permissions`
- The `app` and `fs` params to `initialise` are now grouped together by passing the `permissions` parameter.
- Decrypt `readKey` from auth lobby (behind the scenes)



### v0.17.3

Upgrade to js-ipfs v0.50


### v0.17.2

Connect to signaling server to find your other devices more easily.


### v0.17.1

Upgrade to CIDv1.


### v0.17

###### Changes

- `initialise` now accepts two additional options, named "prerequisites":
   ```javascript
   const { prerequisites, scenario, state } = await wn.initialise({
     // Will ask the user permission to store
     // your apps data in `private/Apps/Nullsoft/Winamp`
     app: {
       name: "Winamp",
       creator: "Nullsoft"
     },

     // Ask the user permission for additional filesystem paths
     fs: {
       privatePaths: [ "Music" ],
       publicPaths: [ "Mixtapes" ]
     }
   })
   ```
- Those prerequisites are passed to the `wn.redirectToLobby` function.  
  (So the auth lobby has the correct parameters to determine the permissions to ask the user)
- Adds the ability to use multiple apps with one file system (closes #73)
- The SDK now handles multiple UCANs
- Works offline (fixes #82)
- Added `initialize` (american spelling) as an alias for `initialise`
- Adds ability to set the potency `ptc` of a UCAN
- Uses the Fission gateway as a [delegate node](https://github.com/ipfs/js-ipfs/blob/2b24f590041a0df9da87b75ae2344232fe22fe3a/docs/CONFIG.md#delegates)

###### Breaking changes

- File system actions (ie. POSIX interface methods) throw an error when they are missing the necessary permissions (read UCAN)
- Does no longer automatically call the `fs.publicise()` method. You have to call this yourself from now on.
- Changed the `fs.appPath` function (see README)
- Renamed `scenario.isAuthenticated` to `scenario.isAuthorised`
- Renamed `scenario.continuum` to `scenario.continuation`
- The first parameter to `redirectToLobby` has now become the second parameter.
- Replaced `deauthenticate` with `leave`, which now redirects to the auth lobby automatically, so you can "sign out" there as well.


### v0.16.x

- Fixed issue with private trees
- Improved connectivity
- Switched from AES-128 read keys to AES-256


### v0.16

- Big rewrite of filesystem
  - private side derives names using bloomfilters and stores nodes in an MMPT
  - reorganize header info on public side and store metadata/skeleton as cbor data
- Improved `fs.write` method, is an alias for `add` now (because `add` overwrites by default)
- Improved file system loading and saving


### v0.15

- Skipped because of a botched npm publish


### v0.14.3

- Added apps API `apps.create`, `apps.index`, `apps.deleteByURL`


### v0.14.2

- Improved DNSLink lookup error handling
- Reduced time-to-save for the file system to 3 seconds instead of 5
- Removed unnecessary `console.log` calls
- Updated default `js-ipfs` to `v0.48.1` (was `v0.48.0`)


### v0.14.1

Removed the default import from the index file. Now you use the SDK as follows, browser stays the same.

```js
import * as sdk from 'fission-sdk'
import { initialise } from 'fission-sdk'

sdk.initialise()
initialise()
```


### v0.14.0

- Renamed `isAuthenticated` to `initialise`
- `initialise` will return an instance of the file system (can be disabled in case you use web workers)
- Adds `loadFileSystem` to load a file system (called automatically from `initialise` unless disabled). This function is responsible for caching the file system locally and making a file system if the user hasn't got one yet.
- Adds the `fs.appPath.private` and `fs.appPath.public` function to build paths.
- Adds the `fs.exists`, `fs.read` and `fs.write` file system methods


src/ipfs/config.ts
import loadScript from 'load-script2'

import { IPFS } from './types'
import { setup } from '../setup/internal'


type IpfsWindow = {
  Ipfs?: { create: (options: unknown) => IPFS }
}


export const JS_IPFS = 'https://cdnjs.cloudflare.com/ajax/libs/ipfs/0.50.1/index.min.js'
export const PEER_WSS = '/dns4/node.fission.systems/tcp/4003/wss/p2p/QmVLEz2SxoNiFnuyLpbXsH6SvjPTrHNMU88vCQZyhgBzgw'
export const SIGNALING_SERVER = '/dns4/webrtc.runfission.com/tcp/443/wss/p2p-webrtc-star/'
export const DELEGATE_ADDR = '/dns4/ipfs.runfission.com/tcp/443/https'


let ipfs: IPFS | null = null


export const defaultOptions = {
  config: {
    Addresses: {
      Delegates: [ DELEGATE_ADDR ],
      Swarm: [ SIGNALING_SERVER ]
    },
    Bootstrap: [ PEER_WSS ]
  },
  preload: {
    enabled: false
  }
}

export const set = (userIpfs: unknown): void => {
  ipfs = userIpfs as IPFS
}

export const get = async (): Promise<IPFS> => {
  if (!ipfs) {
    await loadScript(JS_IPFS)

    const Ipfs = await (window as IpfsWindow).Ipfs
    if (!Ipfs) throw new Error(`Unable to load js-ipfs using the url: \`${JS_IPFS}\``)

    ipfs = await Ipfs.create({
      ...defaultOptions,
      ...setup.ipfs
    })
  }

  await ipfs.swarm.connect(PEER_WSS)

  return ipfs
}
